pkg/ipam/ipam.go

```diff
@@ -130,6 +130,9 @@ func NewIPAM(nodeAddressing types.NodeAddressing, c Configuration, owner Owner,
 		}
 	case ipamOption.IPAMClusterPoolV2:
 		log.Info("Initializing ClusterPool v2 IPAM")
+		if !clientset.IsEnabled() {
+			log.Fatal("ClusterPool v2 IPAM is not compatible with lb-only datamode")
+		}
```

Comment:
I'd suggest to move this before the log.Info. Also, let's say "datapath mode" instead of "datamode":

```suggestion
		if !clientset.IsEnabled() {
			log.Fatal("ClusterPool v2 IPAM is not compatible with lb-only datapath mode")
		}
		log.Info("Initializing ClusterPool v2 IPAM")
```